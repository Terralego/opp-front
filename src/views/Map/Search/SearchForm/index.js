import { withTranslation } from 'react-i18next';

import { connectAppProvider } from '../../../../AppLayout/context';
import { SearchForm } from './SearchForm';

export default connectAppProvider({
  properties: 'searchFormProperties',
  setProperties: 'actions.setSearchFormProperties',
  getFirstPageFilteredViewpoints: 'actions.getFirstPageFilteredViewpoints',
  forceResultUnfolding: 'actions.forceResultUnfolding',
  resetSearchForm: 'actions.resetSearchForm',
  resetMapInitialState: 'actions.resetMapInitialState',
})(withTranslation()(SearchForm));
