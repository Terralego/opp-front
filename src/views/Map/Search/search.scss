@import '../../../variables.scss';

.main__content {
  .search {
    z-index: 8;
    flex: initial;
    height: 100vh;
    width: 300px;
    color: $white;
    margin-left: -17rem;
    transform: translateX(17rem);
    transition: transform $transitionSpeed ease-in;
    font-size: 0.9em;
    background-color: $solidgray;
    box-shadow: 0 0 4px 0 rgba(0, 0, 0, 1);

    &.search_hide {

      transform: translateX(0);

      .search_tabs_title {

        button > span {
          transform: scaleX(-1);
        }
      }

      .bp3-tabs {
        padding: 0 30px;
      }
    }

    .search_tabs_title {
      display: flex;
      text-transform: uppercase;
      justify-content: space-between;
      padding-left: 4px;
      color: $white;
      margin-left: 10px;

      .search__button {
        span {
          color: $white;
          transition: margin $transitionSpeed ease;
        }

        &:hover {
          background-color: #3d4f5c;
        }
      }
    }

    .control-label {
      margin-left: 10px;
    }

    @extend .override-tabs-bp3;
  }
}
