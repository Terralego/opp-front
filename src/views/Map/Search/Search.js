import React from 'react';
import PropTypes from 'prop-types';

import { Button, Tab, Tabs } from '@blueprintjs/core';
import classNames from 'classnames';

import SearchForm from './SearchForm';
import { getSchema } from './utils/schemaForm';
import {
  configAdvancedSearch,
  configSimplesSearch,
} from './utils/configForm';

import translateMock from '../../../utils/translate';
import { fetchFilterOptions } from '../../../services/viewpoints';

import './search.scss';

const ID_SEARCH_PANEL = 'search';
const ID_SEARCH_SIMPLE = 'searchSimple_tab';
const ID_SEARCH_ADVANCED = 'searchAdvanced_tab';

const itemsPerPage = 5;

const getValuesFilters = (schemaForm, { photographers, ...data }) => {
  const cleanedData = {
    ...data,
    photographers: photographers.map(item => ({
      label: item.properties.name,
      value: item.uuid,
    })),
  };
  return schemaForm.map(filter => {
    const values = cleanedData[filter.name] !== undefined
      ? { values: cleanedData[filter.name] }
      : {};
    return { ...filter, ...values };
  });
};

const getFiltersBySearch = (schemaForm, config) => (
  config.map(name => {
    const index = schemaForm.findIndex(filter => filter.name === name);
    return { ...schemaForm[index] };
  })
);

export class Search extends React.PureComponent {
  static propTypes = {
    isSearchUnfold: PropTypes.bool,
    toggleSearchFoldedState: PropTypes.func,
    t: PropTypes.func,
  };

  static defaultProps = {
    isSearchUnfold: false,
    toggleSearchFoldedState () {},
    t: translateMock({
      'map.title': 'Search viewpoints',
      'map.simple-search': 'Simple Search',
      'map.advanced-search': 'Advanced Search',
    }),
  };

  state = {
    navTabId: ID_SEARCH_SIMPLE,
    simplesSearchFilters: [],
    advancedSearchFilters: [],
  };

  componentDidMount () {
    this.getFilters();
  }


  componentWillUnmount () {
    this.isUnmount = true;
  }

  handleNavSearchTabChange = navTabId => this.setState({ navTabId });

  getFilters = async () => {
    try {
      const data = await fetchFilterOptions();
      const schema = getSchema(this.props);
      const filters = await getValuesFilters(schema, data);
      if (this.isUnmount) {
        return;
      }
      this.setState({
        simplesSearchFilters: getFiltersBySearch(filters, configSimplesSearch),
        advancedSearchFilters: getFiltersBySearch(filters, configAdvancedSearch),
      });
    } catch (e) {
      // eslint-disable-next-line no-console
      console.error(e);
    }
  };

  render () {
    const {
      t,
      isSearchUnfold,
      toggleSearchFoldedState,
    } = this.props;
    const {
      navTabId,
      simplesSearchFilters,
      advancedSearchFilters,
    } = this.state;

    return (
      <>
        <div
          className={classNames('search', { search_hide: !isSearchUnfold })}
        >
          <div className="search_tabs_title">
            <h3>{t('map.title')}</h3>
            <Button
              className="search__button"
              onClick={toggleSearchFoldedState}
              icon="arrow-left"
              minimal
            />
          </div>
          <Tabs
            id={ID_SEARCH_PANEL}
            onChange={this.handleNavSearchTabChange}
            selectedTabId={navTabId}
          >
            <Tab
              id={ID_SEARCH_SIMPLE}
              className="search-filters search-simple-content"
              title={t('map.simple-search')}
              panel={(
                <SearchForm
                  itemsPerPage={itemsPerPage}
                  filters={simplesSearchFilters}
                />
              )}
            />

            <Tab
              id={ID_SEARCH_ADVANCED}
              className="search-filters search-advanced-content"
              title={t('map.advanced-search')}
              panel={(
                <SearchForm
                  itemsPerPage={itemsPerPage}
                  filters={advancedSearchFilters}
                />
              )}
            />
          </Tabs>
        </div>
      </>
    );
  }
}

export default Search;
