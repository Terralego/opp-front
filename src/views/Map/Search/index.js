import { withTranslation } from 'react-i18next';

import { Search } from './Search';
import { connectAppProvider } from '../../../AppLayout/context';

export default connectAppProvider({
  map: 'map',
  isSearchUnfold: 'isSearchUnfold',
  toggleSearchFoldedState: 'actions.toggleSearchFoldedState',
})(withTranslation()(Search));
