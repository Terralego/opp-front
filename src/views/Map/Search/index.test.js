import React from 'react';
import renderer from 'react-test-renderer';

import Search from './Search';

jest.mock('./SearchForm', () => () => <p>SearchFormMock</p>);

it('should render', () => {
  const tree = renderer.create(<Search />).toJSON();
  expect(tree).toMatchSnapshot();
});
