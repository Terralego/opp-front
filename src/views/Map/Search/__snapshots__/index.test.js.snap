// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="search search_hide"
>
  <div
    className="search_tabs_title"
  >
    <h3>
      Search viewpoints
    </h3>
    <button
      className="bp3-button bp3-minimal search__button"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      type="button"
    >
      <span
        className="bp3-icon bp3-icon-arrow-left"
        icon="arrow-left"
      >
        <svg
          data-icon="arrow-left"
          height={16}
          viewBox="0 0 16 16"
          width={16}
        >
          <desc>
            arrow-left
          </desc>
          <path
            d="M13.99 6.99H4.41L7.7 3.7a1.003 1.003 0 0 0-1.42-1.42l-5 5a1.014 1.014 0 0 0 0 1.42l5 5a1.003 1.003 0 0 0 1.42-1.42L4.41 8.99H14c.55 0 1-.45 1-1s-.46-1-1.01-1z"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </button>
  </div>
  <div
    className="bp3-tabs"
  >
    <div
      className="bp3-tab-list"
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      role="tablist"
    >
      <div
        className="bp3-tab-indicator-wrapper"
      >
        <div
          className="bp3-tab-indicator"
        />
      </div>
      <div
        aria-controls="bp3-tab-panel_search_searchSimple_tab"
        aria-disabled={false}
        aria-expanded={true}
        aria-selected={true}
        className="bp3-tab search-filters search-simple-content"
        data-tab-id="searchSimple_tab"
        id="bp3-tab-title_search_searchSimple_tab"
        onClick={[Function]}
        role="tab"
        tabIndex={0}
      >
        Simple Search
      </div>
      <div
        aria-controls="bp3-tab-panel_search_searchAdvanced_tab"
        aria-disabled={false}
        aria-expanded={false}
        aria-selected={false}
        className="bp3-tab search-filters search-advanced-content"
        data-tab-id="searchAdvanced_tab"
        id="bp3-tab-title_search_searchAdvanced_tab"
        onClick={[Function]}
        role="tab"
        tabIndex={0}
      >
        Advanced Search
      </div>
    </div>
    <div
      aria-hidden={false}
      aria-labelledby="bp3-tab-title_search_searchSimple_tab"
      className="bp3-tab-panel search-filters search-simple-content"
      id="bp3-tab-panel_search_searchSimple_tab"
      role="tabpanel"
    >
      <p>
        SearchFormMock
      </p>
    </div>
    <div
      aria-hidden={true}
      aria-labelledby="bp3-tab-title_search_searchAdvanced_tab"
      className="bp3-tab-panel search-filters search-advanced-content"
      id="bp3-tab-panel_search_searchAdvanced_tab"
      role="tabpanel"
    >
      <p>
        SearchFormMock
      </p>
    </div>
  </div>
</div>
`;
